### PR TITLE
rgw/admin: remove Debug field from API type

### PR DESCRIPTION
**Pull request by leseb**
Instead of implementing our own logging let's the consumer implement
their own potential debug for each HTTP requests.

Closes: https://github.com/ceph/go-ceph/issues/508
Signed-off-by: Sébastien Han <seb@redhat.com>

## Checklist
- [ ] Added tests for features and functional changes
- [ ] Public functions and types are documented
- [ ] Standard formatting is applied to Go code

### CODE REVIEW

**Review by phlogistonjohn — APPROVED** [review]
LGTM. Thanks a ton for doing this!

File: rgw/admin/README.md [hunk 1]
```diff
@@ -24,9 +24,6 @@ func main() {
         panic(err)
     }
 
-    // To enable debug requests
-    // co.Debug = true
```
Comment (thotz) [inline rgw/admin/README.md:28]: For my understanding, here end-user need to define the debug API interface, right?. Do we need to mention that here?

Comment (leseb) [reply rgw/admin/README.md:28]: Yes, the consumer needs to define a new type for mutating the HTTP client. I'm not sure if it's worth mentioning...

Comment (phlogistonjohn) [reply rgw/admin/README.md:28]: If I was going to mention debugging - I'd just note that the http client type is an interface such that you can mock it or wrap it to fit your needs. And perhaps point to the test code with a hyperlink for an example.

Comment (thotz) [reply rgw/admin/README.md:28]: Thanks @leseb for clarification. I am okay with or without doc changes

**Pull request merged**